src/datepicker/DatePickerState.ts

```diff
@@ -5,49 +5,64 @@
  */
 
 import * as React from "react";
+import { isValid } from "date-fns";
 import { useControllableState } from "@chakra-ui/hooks";
 import {
   useCompositeState,
   usePopoverState,
   unstable_useId as useId,
 } from "reakit";
 
-import { useCalendarState } from "../calendar";
 import { setTime, isInvalid } from "./__utils";
+import { DateValue, useCalendarState } from "../calendar";
 import { useDatePickerFieldState } from "./DatePickerFieldState";
 import { DatePickerStateInitialProps, ValidationState } from "./index.d";
 
 export const useDatePickerState = (props: DatePickerStateInitialProps = {}) => {
   const {
-    value: initialValue,
-    defaultValue = new Date(),
+    value: initialDate,
+    defaultValue: defaultValueProp,
     onChange,
-    minValue,
-    maxValue,
+    minValue: minValueProp,
+    maxValue: maxValueProp,
     isDisabled,
     isReadOnly,
     isRequired,
     pickerId: pickerIdProp,
     dialogId: dialogIdProp,
     formatOptions,
-    placeholderDate,
+    placeholderDate: placeholderDateProp,
   } = props;
 
   const { id: pickerId } = useId({ id: pickerIdProp, baseId: "picker" });
   const { id: dialogId } = useId({ id: dialogIdProp, baseId: "dialog" });
 
+  const defaultValue =
+    defaultValueProp && isValid(defaultValueProp)
+      ? new Date(defaultValueProp)
+      : new Date();
+
   const [value, setValue] = useControllableState({
-    value: initialValue,
+    value: initialDate,
     defaultValue,
     onChange,
     shouldUpdate: (prev, next) => prev !== next,
   });
-  const dateValue = value != null ? new Date(value) : undefined;
+
+  const dateValue = value && isValid(value) ? new Date(value) : undefined;
```

Comment:
Can we do 

```js
const dateValue = isValid(value) ? new Date(value) : undefined;
```

does the isValid handles the undefined argumant?

Reply:
Nope, it doesn't.

Can we write our own isValid with datefn function?

Reply:
Yeah just enhance it, use their isValid function and check for undefined values